### PR TITLE
[#503] Support short topic names in MQTT protocol adapter

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java [hunk 1]
```diff
@@ -490,6 +490,44 @@ public void testOnUnauthenticatedMessageDoesNotSendPubAckOnFailure(final TestCon
         verify(endpoint, never()).publishAcknowledge(anyInt());
     }
 
+    @Test
+    public void testTopicNames(final TestContext ctx) {
```
Comment (sophokles73) [inline adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java]: what is this test supposed to verify? can you please add a JavaDoc comment?

File: adapters/mqtt-vertx/src/test/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapterTest.java [hunk 2]
```diff
@@ -204,6 +204,36 @@ public void testOnPublishedMessageUsesDeviceIdentityForTopicWithoutTenant(final
         assertThat(downstreamAddress.getTenantId(), is("my-tenant"));
     }
 
+    /**
+     * Verifies that the adapter supports all required topic names
+     *
+     * @param ctx The helper to use for running tests on vert.x.
+     */
+    @Test
+    public void testTopicNames(final TestContext ctx) {
```
Comment (sophokles73) [inline adapters/mqtt-vertx/src/test/java/org/eclipse/hono/adapter/mqtt/impl/VertxBasedMqttProtocolAdapterTest.java]: this seems to be a test case for verifying behavior of the *mapTopic* method.
IMHO you should add assertions as to whether the method has actually mapped the message topic to the correct address.

So far we try to reflect the method under test in the test method's name. Maybe something like `testMapTopicSupportsShortAndLongTopicNames`?

File: core/src/main/java/org/eclipse/hono/util/EndpointType.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat Inc and others.
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/util/EndpointType.java]: I don't know about you, but I already live in 2018 ;-)

**Comment by dejanb** [comment]
@sophokles73 These should be fixed now. Thanks for the feedback

**Pull request merged**